src/github.com/getlantern/flashlight/genconfig/genconfig.go

```diff
@@ -221,37 +222,46 @@ func loadTemplate(name string) string {
 	return string(bytes)
 }
 
-func feedDomains() {
+func feedMasquerades() {
 	wg.Add(numberOfWorkers)
 	for i := 0; i < numberOfWorkers; i++ {
 		go grabCerts()
 	}
 
-	for _, domain := range domains {
-		domainsCh <- domain
+	for _, masq := range masquerades {
+		if masq != "" {
+			inputCh <- masq
+		}
 	}
-	close(domainsCh)
+	close(inputCh)
 	wg.Wait()
 	close(masqueradesCh)
 }
 
-// grabCerts grabs certificates for the domains received on domainsCh and sends
+// grabCerts grabs certificates for the masquerades received on masqueradesCh and sends
 // *masquerades to masqueradesCh.
 func grabCerts() {
 	defer wg.Done()
 
-	for domain := range domainsCh {
+	for masq := range inputCh {
+		parts := strings.Split(masq, " ")
+		if len(parts) != 2 {
+			log.Error("Bad line! '" + masq + "'")
+			continue
+		}
+		ip := parts[0]
+		domain := parts[1]
 		_, blacklisted := blacklist[domain]
 		if blacklisted {
 			log.Tracef("Domain %s is blacklisted, skipping", domain)
 			continue
 		}
-		log.Tracef("Grabbing certs for domain: %s", domain)
+		log.Tracef("Grabbing certs for IP %s, domain %s", ip, domain)
 		cwt, err := tlsdialer.DialForTimings(&net.Dialer{
 			Timeout: 10 * time.Second,
-		}, "tcp", domain+":443", false, nil)
+		}, "tcp", ip+":443", false, &tls.Config{ServerName: domain})
```

Comment:
Just that SNI is theoretically easier to block, although it probably doesn't matter too much in this case.